data/wbd/generate_pre_clip_fim_huc8.py

```diff
@@ -0,0 +1,406 @@
+
+import argparse
+import datetime as dt
+import logging
+import os
+import shutil
+import subprocess
+from multiprocessing import Pool
+
+from clip_vectors_to_wbd import subset_vector_layers
+from dotenv import load_dotenv
+
+
+'''
+    Overview:
+      This script was created to absolve run_unit_wb.sh from getting the huc level WBD layer, calling
+      clip_vectors_to_wbd.py, and clipping the WBD for every run, which added a significant amount of
+      processing time for each HUC8. Using this script, we generate the necessary pre-clipped .gpkg files
+      for the rest of the processing steps.
+
+      Read in environment variables from src/bash_variabls.env & config/params_template.env.
+      Parallelize the creation of .gpkg files per HUC:
+        Get huc level WBD layer from National, call the subset_vector_layers function, and clip the wbd.
+        A plethora gpkg files per huc are generated (see args to subset_vector_layers)
+        and placed within the output directory specified as the <outputs_dir> argument.
+
+    Usage:
+        generate_pre_clipped_wbd.py
+            -wbd /data/inputs/wbd/WBD_National_EPSG_5070_WBDHU8_clip_dem_domain.gpkg
+            -n /data/inputs/pre_clip_huc8/24_3_20
+            -u /data/inputs/huc_lists/included_huc8.lst
+            -j 6
+            -o
+
+    Notes:
+      If running this script to generate new data, modify the pre_clip_huc_dir variable in
+      src/bash_variables.env to the corresponding outputs_dir argument after running and testing this script.
+      The newly generated data should be created in a new folder using the format <year_month_day>
+             (i.e. September 26, 2023 would be 23_9_26)
+'''
+
+srcDir = os.getenv('srcDir')
+projectDir = os.getenv('projectDir')
+
+load_dotenv(f'{srcDir}/bash_variables.env')
+load_dotenv(f'{projectDir}/config/params_template.env')
+
+# Variables from src/bash_variables.env
+DEFAULT_FIM_PROJECTION_CRS = os.getenv('DEFAULT_FIM_PROJECTION_CRS')
+inputsDir = os.getenv('inputsDir')
+input_WBD_gdb = os.getenv('input_WBD_gdb')
+input_DEM = os.getenv('input_DEM')
+input_DEM_domain = os.getenv('input_DEM_domain')
+input_nwm_lakes = os.getenv('input_nwm_lakes')
+input_nwm_catchments = os.getenv('input_nwm_catchments')
+input_NLD = os.getenv('input_NLD')
+input_levees_preprocessed = os.getenv('input_levees_preprocessed')
+input_GL_boundaries = os.getenv('input_GL_boundaries')
+input_nwm_flows = os.getenv('input_nwm_flows')
+input_nwm_headwaters = os.getenv('input_nwm_headwaters')
+input_nld_levee_protected_areas = os.getenv('input_nld_levee_protected_areas')
+
+# Variables from config/params_template.env
+wbd_buffer = os.getenv('wbd_buffer')
+wbd_buffer_int = int(wbd_buffer)
+
+
+def __setup_logger(outputs_dir):
+    '''
+    Set up logging to file. Since log file includes the date, it will be overwritten if this
+    script is run more than once on the same day.
+    '''
+
+    curr_date = dt.datetime.now().strftime("%m_%d_%Y")
```

Comment:
A couple of things:
1. I think we might prefer to use `datetime.utcnow()` here (or perhaps `datetime.now(timezone.utc)` is even better). Also applies to lines 139, 187, etc.
2.  It might be good to first define a variable such as `datetime_now = dt.datetime.now(timezone.utc)`, and then `curr_date = datetime_now.strftime("%m_%d_%Y")` (and likewise for line 97) which would keep the datetime stamp synchronized. Also applies to line 188.